### PR TITLE
[HW11] Add `UpdateProject` mutation

### PR DESCRIPTION
**Pull request by Raily01**
### Summary

A brief description of the pull request.

### How it works

Screenshots/screencasts of the pull request introduced functionality.

### Test plan

List of steps to manually test introduced functionality:

- Go to http://localhost:3000/graphiql
- Make request using schema:
```graphql
  query {
    me {
      id
      name
    }
  }
```

### Deploy notes

Notes regarding deployment the contained body of work.
These should note any db migrations, services, scripts, etc.

### Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My pull request is shorter than 500 lines

### CODE REVIEW

**Review by w1nway — COMMENTED** [review]
Good Job, my friend

File: app/graphql/mutations/update_project.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  class UpdateProject < BaseMutation
+    include GraphqlErrors
+
+    argument :input, Types::Inputs::UpdateProjectInput, required: true
+
+    type Types::Payloads::UpdateProjectPayload
+
+    def resolve(input:)
+      project = Project.find(input.id)
+      result = Projects::Update.call(project: project, project_params: input.to_h, user: current_user)
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/update_project.rb:11]:
```suggestion
      result = Projects::Update.call(project: project, project_params: input.to_h.except(:id), user: current_user)
```

File: app/graphql/types/inputs/update_project_input.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module Types
+  module Inputs
+    class UpdateProjectInput < Types::BaseInputObject
+      # argument :Projcect
+      argument :id, ID, required: true
+      argument :name, String, required: true
```
Comment (RasimKhusaenov) [inline app/graphql/types/inputs/update_project_input.rb:6]: Why it is required?

File: app/graphql/mutations/update_project.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+      if result.success?
+        result.to_h
+      else
+        result.to_h.merge(errors: formatted_errors(result.project))
+      end
```
Comment (RasimKhusaenov) [inline app/graphql/mutations/update_project.rb:17]:
```suggestion
      result.to_h.merge(errors: formatted_errors(result.project))
```

**Pull request closed without merging**